pynest/nest/spatial/hl_api_spatial.py

```diff
@@ -83,7 +87,11 @@ def n(dimension):
         return CreateParameter('distance', {'dimension': dimension})
 
 
-distance = DistanceParameter()
+@functools.cache
+def __getattr__(name):
+    if name == "distance":
+        return DistanceParameter()
+    raise AttributeError(f"module {__name__} has no attribute {name}")
```

Comment:
This `__getattr__` call should be moved up one module to `nest.spatial`, otherwise the import in `nest.spatial` triggers this import at import-time again. I propose this:

### In `nest/spatial/__init__.py`

```python
from .hl_api_spatial import _distance_singleton

__all__ = [..., "distance"]

def __getattr__(name):
  if name == "distance":
    return _distance_singleton()
```

### In `hl_api_spatial.py`

```python
@functools.cache
def _distance_singleton():
  return DistanceParameter()
```